Multichannl images in Segment fails due to drawcontours

### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.


### Ultralytics YOLO Component

Train

### Bug

I updated to newest version to double check its still a problem , 8.4.8 .
I have 30 channel images, and they are saved as mulltipage TIFF is directed in your docs. data.yaml has channels: 30. and the models architecture shows as 30 inputs. 
so good so far. 
In read_images_and_labels -> augment -> drawContours, I get an assertion about drawcontours drawing on those channels:

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Anaconda3\envs\pytorch2\Scripts\yolo.exe\__main__.py", line 7, in <module>
  File "C:\Anaconda3\envs\pytorch2\Lib\site-packages\ultralytics\cfg\__init__.py", line 986, in entrypoint
    getattr(model, mode)(**overrides)  # default args from model
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Anaconda3\envs\pytorch2\Lib\site-packages\ultralytics\engine\model.py", line 774, in train
    self.trainer.train()
  File "C:\Anaconda3\envs\pytorch2\Lib\site-packages\ultralytics\engine\trainer.py", line 244, in train
    self._do_train()
  File "C:\Anaconda3\envs\pytorch2\Lib\site-packages\ultralytics\engine\trainer.py", line 405, in _do_train
    for i, batch in pbar:
  File "C:\Anaconda3\envs\pytorch2\Lib\site-packages\ultralytics\utils\tqdm.py", line 350, in __iter__
    for item in self.iterable:
  File "C:\Anaconda3\envs\pytorch2\Lib\site-packages\ultralytics\data\build.py", line 76, in __iter__
    yield next(self.iterator)
          ^^^^^^^^^^^^^^^^^^^
  File "C:\Anaconda3\envs\pytorch2\Lib\site-packages\torch\utils\data\dataloader.py", line 631, in __next__
    data = self._next_data()
           ^^^^^^^^^^^^^^^^^
  File "C:\Anaconda3\envs\pytorch2\Lib\site-packages\torch\utils\data\dataloader.py", line 675, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Anaconda3\envs\pytorch2\Lib\site-packages\torch\utils\data\_utils\fetch.py", line 51, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Anaconda3\envs\pytorch2\Lib\site-packages\torch\utils\data\_utils\fetch.py", line 51, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
            ~~~~~~~~~~~~^^^^^
  File "C:\Anaconda3\envs\pytorch2\Lib\site-packages\ultralytics\data\base.py", line 376, in __getitem__
    return self.transforms(self.get_image_and_label(index))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Anaconda3\envs\pytorch2\Lib\site-packages\ultralytics\data\augment.py", line 187, in __call__
    data = t(data)
           ^^^^^^^
  File "C:\Anaconda3\envs\pytorch2\Lib\site-packages\ultralytics\data\augment.py", line 187, in __call__
    data = t(data)
           ^^^^^^^
  File "C:\Anaconda3\envs\pytorch2\Lib\site-packages\ultralytics\data\augment.py", line 1715, in __call__
    return self._transform(labels)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Anaconda3\envs\pytorch2\Lib\site-packages\ultralytics\data\augment.py", line 1761, in _transform
    cv2.drawContours(im_new, instances2.segments[[j]].astype(np.int32), -1, (1, 1, 1), cv2.FILLED)
cv2.error: OpenCV(4.11.0) D:\a\opencv-python\opencv-python\opencv\modules\core\src\copy.cpp:71: error: (-215:Assertion failed) cn <= 4 in function 'cv::scalarToRawData'

### Environment

Ultralytics 8.4.8  Python-3.11.0 torch-2.3.1+cu118 CUDA:0 (NVIDIA RTX A3000 Laptop GPU, 6144MiB)
Setup complete  (16 CPUs, 63.7 GB RAM, 881.5/931.5 GB disk)

OS                     Windows-10-10.0.26100-SP0
Environment            Windows
Python                 3.11.0
Install                pip
Path                   C:\Anaconda3\envs\pytorch2\Lib\site-packages\ultralytics
RAM                    63.71 GB
Disk                   881.5/931.5 GB
CPU                    11th Gen Intel Core(TM) i7-11850H 2.50GHz
CPU count              16
GPU                    NVIDIA RTX A3000 Laptop GPU, 6144MiB
GPU count              1
CUDA                   11.8

numpy                   1.26.4>=1.23.0
matplotlib              3.10.0>=3.3.0
opencv-python           4.11.0.86>=4.6.0
pillow                  10.4.0>=7.1.2
pyyaml                  6.0.2>=5.3.1
requests                2.32.3>=2.23.0
scipy                   1.15.1>=1.4.1
torch                   2.3.1+cu118<2.10,>=1.8.0
torch                   2.3.1+cu118!=2.4.0,<2.10,>=1.8.0; sys_platform == "win32"
torchvision             0.18.1+cu118>=0.9.0
psutil                  5.9.8>=5.8.0
polars                  1.36.1>=0.20.0
ultralytics-thop        2.0.18>=2.0.18

### Minimal Reproducible Example

Try training segment mode on multichannel images (channels over 4). detect mode may not fail because it will not call "drawContours". 

### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

--- a/ultralytics/__init__.py
+++ b/ultralytics/__init__.py
@@ -1,6 +1,6 @@
 # Ultralytics 🚀 AGPL-3.0 License - https://ultralytics.com/license
 
-__version__ = "8.4.8"
+__version__ = "8.4.9"
 
 import importlib
 import os

--- a/ultralytics/data/augment.py
+++ b/ultralytics/data/augment.py
@@ -1745,7 +1745,7 @@ class CopyPaste(BaseMixTransform):
         instances.convert_bbox(format="xyxy")
         instances.denormalize(w, h)
 
-        im_new = np.zeros(im.shape, np.uint8)
+        im_new = np.zeros(im.shape[:2], np.uint8)
         instances2 = labels2.pop("instances", None)
         if instances2 is None:
             instances2 = deepcopy(instances)
@@ -1758,7 +1758,7 @@ class CopyPaste(BaseMixTransform):
         for j in indexes[: round(self.p * n)]:
             cls = np.concatenate((cls, labels2.get("cls", cls)[[j]]), axis=0)
             instances = Instances.concatenate((instances, instances2[[j]]), axis=0)
-            cv2.drawContours(im_new, instances2.segments[[j]].astype(np.int32), -1, (1, 1, 1), cv2.FILLED)
+            cv2.drawContours(im_new, instances2.segments[[j]].astype(np.int32), -1, 1, cv2.FILLED)
 
         result = labels2.get("img", cv2.flip(im, 1))  # augment segments
         if result.ndim == 2:  # cv2.flip would eliminate the last dimension for grayscale images

